cmd/ci-operator-configresolver/main.go

```diff
@@ -119,6 +119,99 @@ func resolveConfig(configAgent agents.ConfigAgent, registryAgent agents.Registry
 	}
 }
 
+const (
+	injectFromOrgQuery     = "injectTestFromOrg"
+	injectFromRepoQuery    = "injectTestFromRepo"
+	injectFromBranchQuery  = "injectTestFromBranch"
+	injectFromVariantQuery = "injectTestFromVariant"
+	injectTestQuery        = "injectTest"
+)
+
+func injectTestFromQuery(w http.ResponseWriter, r *http.Request) (api.Metadata, string, error) {
+	var metadata api.Metadata
+	var test string
+
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusNotImplemented)
+		err := fmt.Errorf("expected GET, got %s", r.Method)
+		if _, errWrite := w.Write([]byte(http.StatusText(http.StatusNotImplemented))); errWrite != nil {
+			err = fmt.Errorf("%s and writing the response body failed with %w", err.Error(), errWrite)
+		}
+		return metadata, test, err
+	}
+	metadata.Org = r.URL.Query().Get(injectFromOrgQuery)
```

Comment:
```go
var meta api.Metadata
for query, field := range map[string]*string{
    injectFromOrgQuery: &meta.Org,
    injectFromRepoQuery: &meta.Repo,
// ...
} {
    value := r.URL.Query().Get(query)
    if value == "" {
        webreg.MissingQuery(w, query)
        return metadata, test, fmt.Errorf("missing query %s", query)
    }
    *field = value
}

Reply:
I based this method on a copy of an existing method, so I addressed both places in a separate commit. Thanks!